domain/src/main/java/com/example/homework3/domain/repository/UserRemoteRepository.kt

```diff
@@ -0,0 +1,11 @@
+package com.example.homework3.domain.repository
+
+import com.example.homework3.domain.model.Item
+import com.example.homework3.domain.model.UserDetails
+
+interface UserRemoteRepository {
+
+    suspend fun getUsers(lastId: Int): Result<List<Item.User>>
```

Comment:
у тебя мешаются элементы с ui репозиторием - Item это элемент для отображения
```suggestion
    suspend fun getUsers(lastId: Int): Result<List<User>>
```